### ISSUE TITLE
syslog_output from Wazuh1 to Wazuh2 receiving nothing

### ISSUE THREAD

**Issue by Serva95** [wazuh/wazuh#8961]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.1.5-4107 | remote-syslog | all-in-one Manager | step-by-step | ubuntu cloud 20.04 updated |

**Description**

Hello team, 

at our security team we are testing a particular wazuh configuration.
We have a "DMZ OT" network where a first instance of wazuh is running which performs agentless monitoring on an network named "OT".
On a second "DMZ IT" network there is a second instance of wazuh that monitors an "IT" network and should receive the alerts raised by the wazuh OT. 
We linked this two instances by clustering them: the wazuh OT is a worker and wazuh IT is the master, this has been done for remote configuration purpose.
At first we configured the wazuh OT to receive data via syslog, we tested this with suricata and it works well. 
After this we configured wazuh OT to output all the alerts through <syslog_output> to the wazuh IT. 
At this point the problems arise: we tried some ssh login fail in the wazuh OT machine, and in the wazuh OT dashboard we can see them but not in the wazuh IT dashboard. 
So we checked that syslog was running correctly through the same test we did with suricata earlier but in the wazuh IT and it worked. 
At this point we stopped syslog in wazuh and opened a syslog server in the wazuh IT machine, with output monitored in console, to test if the data were received. We did the same ssh login fail oln the wazuh OT machine, and the data arrived to the wazuh IT machine through syslog.
At this point we went back to the old configuration with syslog from wazuh and monitored the /var/ossec/log/archive/archive.log to see if the data were received, but nothing was recorded in the file from the syslog.
We saw that the ssh login fail level is 5 and multiple login failed is 10 so wazuh should see them because minimum level is set to 3.

This is the ossec.conf file of wazuh OT:
```xml
  <global>
    <jsonout_output>yes</jsonout_output>
    <alerts_log>yes</alerts_log>
    <logall>yes</logall>
    <logall_json>no</logall_json>
    <email_notification>no</email_notification>
    <smtp_server>smtp.example.wazuh.com</smtp_server>
    <email_from>ossecm@example.wazuh.com</email_from>
    <email_to>recipient@example.wazuh.com</email_to>
    <email_maxperhour>12</email_maxperhour>
    <email_log_source>alerts.log</email_log_source>
    <agents_disconnection_time>10m</agents_disconnection_time>
    <agents_disconnection_alert_time>0</agents_disconnection_alert_time>
  </global>
  
  <cluster>
    <name>wazuh</name>
    <node_name>worker_OT</node_name>
    <node_type>worker</node_type>
    <key>32char</key>
    <port>1516</port>
    <bind_addr>0.0.0.0</bind_addr>
    <nodes>
      <node>192.168.42.10</node>
    </nodes>
    <hidden>no</hidden>
    <disabled>no</disabled>
  </cluster>
  
  <syslog_output>
    <server>192.168.42.10</server>
    <level>3</level>
    <format>json</format>
  </syslog_output>

  <alerts>
    <log_alert_level>3</log_alert_level>
    <email_alert_level>12</email_alert_level>
  </alerts>

  <logging>
    <log_format>plain</log_format>
  </logging>

  <remote>
    <connection>secure</connection>
    <port>1514</port>
    <protocol>tcp</protocol>
    <queue_size>131072</queue_size>
  </remote>
  
  <remote>
    <connection>syslog</connection>
    <port>514</port>
    <protocol>udp</protocol>
    <allowed-ips>192.168.43.0/24</allowed-ips>
    <local_ip>192.168.101.10</local_ip>
  </remote>
  ```
----------------------------------------------------------------------------------------
  
This is the ossec.conf file of wazuh IT:
```xml
  <cluster>
    <name>wazuh</name>
    <node_name>manager_IT</node_name>
    <node_type>master</node_type>
    <key>32char</key>
    <port>1516</port>
    <bind_addr>192.168.42.10</bind_addr>
    <nodes>
      <node>192.168.42.10</node>
    </nodes>
    <hidden>no</hidden>
    <disabled>no</disabled>
  </cluster>
  
  <global>
    <jsonout_output>yes</jsonout_output>
    <alerts_log>yes</alerts_log>
    <logall>yes</logall>
    <logall_json>yes</logall_json>
    <email_notification>no</email_notification>
    <smtp_server>smtp.example.wazuh.com</smtp_server>
    <email_from>ossecm@example.wazuh.com</email_from>
    <email_to>recipient@example.wazuh.com</email_to>
    <email_maxperhour>12</email_maxperhour>
    <email_log_source>alerts.log</email_log_source>
    <agents_disconnection_time>10m</agents_disconnection_time>
    <agents_disconnection_alert_time>0</agents_disconnection_alert_time>
  </global>

  <alerts>
    <log_alert_level>3</log_alert_level>
    <email_alert_level>12</email_alert_level>
  </alerts>

  <logging>
    <log_format>plain</log_format>
  </logging>

  <remote>
    <connection>secure</connection>
    <port>1514</port>
    <protocol>tcp</protocol>
    <queue_size>131072</queue_size>
  </remote>
  
  <remote>
    <connection>syslog</connection>
    <port>514</port>
    <protocol>udp</protocol>
    <allowed-ips>0.0.0.0/0</allowed-ips>
    <local_ip>192.168.42.10</local_ip>
  </remote>
```

As can be seen from here, these are the screenshots of both instances of wazuh
![Screenshot from 2021-06-10 17-51-43](https://user-images.githubusercontent.com/43950391/121656200-8bdc1800-ca9f-11eb-939e-e948e01031b6.png)

![Screenshot from 2021-06-11 10-26-19](https://user-images.githubusercontent.com/43950391/121656245-94345300-ca9f-11eb-892d-916333e73438.png)

![Screenshot from 2021-06-11 10-36-49](https://user-images.githubusercontent.com/43950391/121657783-ffcaf000-caa0-11eb-990b-eefbb897ec3a.png)

-----------------------------------------------------

Do you have any suggestion on how to resolve this problem or an alternative way to send the alert from the first wazuh to another?
Thanks in advance.

**Comment by juliancnn** [comment]
Hi @Serva95 ,

There is currently an issue related to this. You could see it here: **Alert forwarding is not parsed #2499**
When the syslog message arrives with the alert from another manager, it is processed by Wazuh-Analysisd and discarded, without storing it in the `archive.json/log`. (This only happens if the syslog message is an alert).

I am working on a workaraound that may be useful for you. Does each Wazuh Manager own an instance of elasticsearch? or do they use a shared instance?

Regards,
Julian.

**Comment by Serva95** [comment]
Hello, thank you for your answer. 
Each wazuh has its own instance of elasticsearch. 
By the way, in the meantime, I found an alternative way to connect the two different instances, because my team needed this link. If this can help you, I've done it through external integration in the wazuh_OT instance and sent through syslog to wazuh_IT instance that correctly process them.

This is the code of the integration:
```python
#!/usr/bin/python3

import sys
import json
import logging
import logging.handlers

alert_file = open(sys.argv[1])

# Read the alert file
alert_json = json.loads(alert_file.read())
alert_file.close()

# Create a syslog logger for forwarding alerts
logger = logging.getLogger('wazuh_syslog_logger')
logger.setLevel(logging.DEBUG)
handler = logging.handlers.SysLogHandler(address = ('IP_OF_REMOTE_WAZUH', 514), facility="local5")
logger.addHandler(handler)

# Keep only the "full_log" of the alert
out_log=str(alert_json['full_log'])

# Send the alert
logger.warning(out_log)
```

As you can see from the screenshot attached the alerts are correctly forwarded from OT to IT (the wazuh_OT IP is 192.168.101.10)

Wazuh OT instance:
![Screenshot from 2021-06-16 09-45-16](https://user-images.githubusercontent.com/43950391/122180257-8b20f880-ce88-11eb-8c74-c882a7bb78b3.png)

Wazuh IT instance:
![Screenshot from 2021-06-16 09-45-02](https://user-images.githubusercontent.com/43950391/122180275-9116d980-ce88-11eb-8d63-67bd8386b24e.png)

Thanks again for the answer.
Best regards,
Serva.

**Comment by juliancnn** [comment]
Hi @Serva95,

Glad to hear you found a solution!

There is another alternative.  You might want to know.
You currently have a structure like the following:

![Untitled drawing](https://user-images.githubusercontent.com/1891958/122244521-54081280-ce9b-11eb-9f4c-73fa30c81b92.jpg)

#### Reviewing the operation of the Wazuh environment:

Wazuh Manager, receives logs from different sources.
The logs are processed by Wazuh-Analysisd, and in the case of generating alerts, these are stored in the `alert.json` file.
Filebeat, watches the `alert.json` file, reads the alerts and sends them to Elasticsearch.
Finally, Wazuh-App (application inside Kibana), is able to communicate with the manager through its API and also to collect alert data from Elasticsearch.

This is a brief summary, abstracting from details, in order to explain the workaraund (You can learn more about the architecture and operation of wazuh [here](https://documentation.wazuh.com/current/getting-started/architecture.html)).

Another possible solution is to use an architecture like the following:

![draw2](https://user-images.githubusercontent.com/1891958/122246160-b44b8400-ce9c-11eb-83ee-25ab5513e9d5.jpg)


The alternative is to use a second instance of Filebeat, which reads the alerts and sends them directly to the Elasticsearch on the other host.

Advantages:
- No loss of log source information (location field).
- Logs are processed only once, with the OT Manager ruleset.
- Logs are processed in a single Manager,

Disadvantages:
- It is a bit more complex than the script that forwards the logs.

The solution could be carried out in 3 stages:
1) Configure Elasticsearch in IT to accept connections from remote hosts.
2) Configure a second instance of Filebeat on the OT server
3) Configure Wazuh app on the IT server to connect to the Wazuh manager API on OT


### Configure Elasticsearch in IT to accept connections from remote hosts.
If you have the default configuration, elasticsearch will only accept connections from the local host. To change this you should edit the `/etc/elasticsearch/elasticsearch.yml` file and replace the line:

```
network.host: 127.0.0.1
```
 with:
```
network.host: 0.0.0.0
```
Then restart the service:

```
systemctl restart elasticsearch.service
```

### Configure a second instance of Filebeat on the OT server
For this, we will create empty folders for the new instance to use and then clone the current instance filebealbeat

```
# Create empty folders for the new instance
mkdir  /var/lib/filebeat-remote
mkdir  /var/log/filebeat-remote
# Clone the filebeat instance
cp -a /etc/filebeat /etc/filebeat-remote
```
#### Configuration of the new instance (remote Filebeat)

To do this, we edit the `/etc/filebeat-remote/filebeat.yml` configuration file, update the certificate path, point it to the IT elastic and disable certificate verification.

```
vim /etc/filebeat-remote/filebeat.yml
```
It could look like the following:

```yml
# Wazuh - Filebeat configuration file
output.elasticsearch:
  # IP of the IT server (where we will send the OT alerts)
  hosts: ["192.168.0.35:9200"]
  protocol: https
  # User and password for elasticsearch, by default in the installation is admin:admin, 
  # I recommend to change it once everything is working correctly.
  username: "admin"
  password: "admin"
  # Add this option to disable the verification of certificates (This does not disable encryption).
  ssl.verification_mode: none
  # Change the certificates path from /etc/filebeat/ to /etc/filebeat-remote/
  ssl.certificate_authorities:
    - /etc/filebeat-remote/certs/root-ca.pem
  ssl.certificate: "/etc/filebeat-remote/certs/filebeat.pem"
  ssl.key: "/etc/filebeat-remote/certs/filebeat.key"
setup.template.json.enabled: true
setup.template.json.path: '/etc/filebeat/wazuh-template.json'
setup.template.json.name: 'wazuh'
setup.ilm.overwrite: true
setup.ilm.enabled: false

filebeat.modules:
  - module: wazuh
    alerts:
      enabled: true
    archives:
      enabled: false
```

Finally, it is necessary to bring the certificates from the IT server to the OT server. For this we copy the following files from the IT server:

| origin | destination |
|---------|-----------------|
|IT:/etc/filebeat/certs/root-ca.pem | OT:/etc/filebeat-remote/certs/root-ca.pem |
|IT:/etc/filebeat/certs/filebeat.pem | OT:/etc/filebeat-remote/certs/filebeat.pem |
|IT:/etc/filebeat/certs/filebeat.key | OT:/etc/filebeat-remote/certs/filebeat.key |

#### Configuration of the service (remote Filebeat)
Now, we only have to configure the startup of the filebeat-remote service. To do this, we start from the configuration of the instance already installed, cloning its service script

```
# Clone the service script
cp /lib/systemd/system/filebeat.service /lib/systemd/system/filebeat-remote.service
```

Then we edit the script, changing some `/etc/filebeat/` paths to `/etc/filebeat-remote/`. It is important to note that the home path does not change, so it should look like the following:

```
[Unit]
Description=Filebeat sends log files to Logstash or directly to Elasticsearch.
Documentation=https://www.elastic.co/products/beats/filebeat
Wants=network-online.target
After=network-online.target

[Service]

Environment="BEAT_LOG_OPTS="
Environment="BEAT_CONFIG_OPTS=-c /etc/filebeat-remote/filebeat.yml"
Environment="BEAT_PATH_OPTS=--path.home /usr/share/filebeat --path.config /etc/filebeat-remote --path.data /var/lib/filebeat-remote --path.logs /var/log/filebeat-remote"
ExecStart=/usr/share/filebeat/bin/filebeat --environment systemd $BEAT_LOG_OPTS $BEAT_CONFIG_OPTS $BEAT_PATH_OPTS
Restart=always

[Install]
WantedBy=multi-user.target
```

Then we reload the list of services and start our second instance of Filbeat:
```
systemctl daemon-reload
systemctl start filebeat-remote
systemctl enable filebeat-remote
```

At this point, the **new alerts** generated in OT should arrive at the elasticsearch instance in IT.

Before proceeding further, we should check if this worked correctly.
To do this, forces the generation of an alert on the OT server, e.g. with failed ssh attempt, then go to `Kibana -> Discover` (in IT host)

![Untitled drawing(1)](https://user-images.githubusercontent.com/1891958/122268372-5c1f7c80-ceb2-11eb-9971-a4228953c7d1.png)

Then select to view the alerts index

![image](https://user-images.githubusercontent.com/1891958/122268669-bfa9aa00-ceb2-11eb-8263-71d56cab1b76.png)

Finally in the right panel you should see the alert.


###  Configure Wazuh app on the IT server to connect to the Wazuh manager API on OT
You may have noticed that the alerts are not yet displayed on the `wazuh app` but on `kibana->discover`.
This is because we have not yet added to the wazuh app, from the IT server, the connection to the wazuh manager in OT.
For this, you must edit the APP configuration file on the IT server (`/usr/share/kibana/optimize/wazuh/config/wazuh.yml`).

You could add the connection to the Manager API in OT, the file should look like the following:

```yml
hosts:
  - managerIT:
     url: https://localhost
     port: 55000
     username: wazuh-wui
     password: wazuh-wui
  - managerOT:
     url: https://192.168.101.10
     port: 55000
     username: wazuh-wui
     password: wazuh-wui

```


Then, when using the wazuh application, you can select which API you want to use, this will allow you to switch between boards of the different managers.

![Untitled drawing(1)](https://user-images.githubusercontent.com/1891958/122271434-c980dc80-ceb5-11eb-81bd-5cb90be9ef49.jpg)


I hope you find this useful, at least informative :). 

Regards,
Julian

**Comment by Serva95** [comment]
Hello Julian, 
thank you very much for all the info provided. This is really helpful !
I have applied all the procedure you recommended and now everything works perfectly. The integration to configure Wazuh app on the IT server to connect to the Wazuh manager API on OT is also a great solution.
Thank you so much for everything.
I also think that the information contained in this issue could be useful to others and I would like to ask if it is possible to integrate it into the official documentation; if you want I can also write some sort of documentation and send you the result.

Best regards,
Serva.

**Comment by juliancnn** [comment]
hi @Serva95 ,

Thank you very much for your contribution to the Wazuh project. I appreciate your collaboration, I have currently created an issue in the documentation repository (wazuh/wazuh-documentation#3974) to comment on the case and see analyze how to proceed with your proposal.

Best regards,
Julian.

**Comment by Serva95** [comment]
Hello Julian, 
thank you for the new issue created.
I thank you again for all the information you have provided me in this issue. Now I close this since my problems have been solved. I remain, of course, available regarding the issue on the documentation.

Best regards,
Serva.

**Comment by MalekBerrich** [comment]
> Hi @Serva95,
> 
> Glad to hear you found a solution!
> 
> There is another alternative. You might want to know. You currently have a architecture like the following:
> 
> ![Untitled drawing](https://user-images.githubusercontent.com/1891958/122244521-54081280-ce9b-11eb-9f4c-73fa30c81b92.jpg)
> 
> #### Reviewing the operation of the Wazuh environment
> Wazuh Manager, receives logs from different sources. The logs are processed by Wazuh-Analysisd, and in the case of generating alerts, these are stored in the `alert.json` file. Filebeat, watches the `alert.json` file, reads the alerts and sends them to Elasticsearch. Finally, Wazuh-App (application inside Kibana), is able to communicate with the manager through its API and also to collect alert data from Elasticsearch.
> 
> This is a brief summary, abstracting from details, in order to explain the workaraund (You can learn more about the architecture and operation of wazuh [here](https://documentation.wazuh.com/current/getting-started/architecture.html)).
> 
> ### Workaround
> Another possible solution is to use an architecture like the following:
> 
> ![draw2](https://user-images.githubusercontent.com/1891958/122246160-b44b8400-ce9c-11eb-83ee-25ab5513e9d5.jpg)
> 
> This solution is based on the use of a second instance of Filebeat, which reads the alerts and sends them directly to the Elasticsearch on the other host.
> 
> Advantages:
> 
> * No loss of log source information (location field).
> * Logs are processed only once, with the OT Manager ruleset.
> * Logs are processed in a single Manager,
> 
> Disadvantages:
> 
> * It is a bit more complex than the script that forwards the logs.
> 
> The solution could be carried out in 3 stages:
> 
> 1. Configure Elasticsearch in IT to accept connections from remote hosts.
> 2. Configure a second instance of Filebeat on the OT server
> 3. Configure Wazuh app on the IT server to connect to the Wazuh manager API on OT
> 
> ### Configure Elasticsearch in IT to accept connections from remote hosts.
> If you have the default configuration, elasticsearch will only accept connections from the local host. To change this behavior you should edit the `/etc/elasticsearch/elasticsearch.yml` file and replace the line:
> 
> ```
> network.host: 127.0.0.1
> ```
> 
> with:
> 
> ```
> network.host: 0.0.0.0
> ```
> 
> Then restart the service:
> 
> ```
> systemctl restart elasticsearch.service
> ```
> 
> > you could read more about it [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-network.html#common-network-settings)
> 
> ### Configure a second instance of Filebeat on the OT server
> For this, we will create empty folders for the new instance to use and then clone the current instance filebealbeat
> 
> ```
> # Create empty folders for the new instance
> mkdir  /var/lib/filebeat-remote
> mkdir  /var/log/filebeat-remote
> # Clone the filebeat instance
> cp -a /etc/filebeat /etc/filebeat-remote
> ```
> 
> #### Configuration of the new instance (remote Filebeat)
> To do this, we edit the `/etc/filebeat-remote/filebeat.yml` configuration file, update the certificate path, point it to the IT elastic and disable certificate verification.
> 
> ```
> vim /etc/filebeat-remote/filebeat.yml
> ```
> 
> It could look like the following:
> 
> ```yaml
> # Wazuh - Filebeat configuration file
> output.elasticsearch:
>   # IP of the IT server (where we will send the OT alerts)
>   hosts: ["192.168.0.35:9200"]
>   protocol: https
>   # User and password for elasticsearch, by default in the installation is admin:admin, 
>   # I recommend to change it once everything is working correctly.
>   username: "admin"
>   password: "admin"
>   # Add this option to disable the verification of certificates (This does not disable encryption).
>   ssl.verification_mode: none
>   # Change the certificates path from /etc/filebeat/ to /etc/filebeat-remote/
>   ssl.certificate_authorities:
>     - /etc/filebeat-remote/certs/root-ca.pem
>   ssl.certificate: "/etc/filebeat-remote/certs/filebeat.pem"
>   ssl.key: "/etc/filebeat-remote/certs/filebeat.key"
> setup.template.json.enabled: true
> setup.template.json.path: '/etc/filebeat/wazuh-template.json'
> setup.template.json.name: 'wazuh'
> setup.ilm.overwrite: true
> setup.ilm.enabled: false
> 
> filebeat.modules:
>   - module: wazuh
>     alerts:
>       enabled: true
>     archives:
>       enabled: false
> ```
> 
> Finally, it is necessary to copy the certificates from the IT server to the OT server. For this we copy the following files from the IT server to OT server:
> 
> origin	destination
> IT:/etc/filebeat/certs/root-ca.pem	OT:/etc/filebeat-remote/certs/root-ca.pem
> IT:/etc/filebeat/certs/filebeat.pem	OT:/etc/filebeat-remote/certs/filebeat.pem
> IT:/etc/filebeat/certs/filebeat.key	OT:/etc/filebeat-remote/certs/filebeat.key
> #### Configuration of the service (remote Filebeat)
> Now, we only have to configure the startup of the filebeat-remote as service. To do this, we start from the configuration of the instance already installed, cloning its service script
> 
> ```
> # Clone the service script
> cp /lib/systemd/system/filebeat.service /lib/systemd/system/filebeat-remote.service
> ```
> 
> Then we edit the script, changing some `/etc/filebeat/` paths to `/etc/filebeat-remote/`. It is important to note that the home path does not change, so it should look like the following:
> 
> ```
> [Unit]
> Description=Filebeat sends log files to Logstash or directly to Elasticsearch.
> Documentation=https://www.elastic.co/products/beats/filebeat
> Wants=network-online.target
> After=network-online.target
> 
> [Service]
> 
> Environment="BEAT_LOG_OPTS="
> Environment="BEAT_CONFIG_OPTS=-c /etc/filebeat-remote/filebeat.yml"
> Environment="BEAT_PATH_OPTS=--path.home /usr/share/filebeat --path.config /etc/filebeat-remote --path.data /var/lib/filebeat-remote --path.logs /var/log/filebeat-remote"
> ExecStart=/usr/share/filebeat/bin/filebeat --environment systemd $BEAT_LOG_OPTS $BEAT_CONFIG_OPTS $BEAT_PATH_OPTS
> Restart=always
> 
> [Install]
> WantedBy=multi-user.target
> ```
> 
> Then we reload the list of services and start our second instance of Filbeat:
> 
> ```
> systemctl daemon-reload
> systemctl start filebeat-remote
> systemctl enable filebeat-remote
> ```
> 
> At this point, the **new alerts** generated in OT should arrive at the elasticsearch instance in IT.
> 
> Before proceeding further, we should check if this worked correctly. To do this, forces the generation of an alert on the OT server, e.g. with failed ssh attempt, then go to `Kibana -> Discover` (in IT host)
> 
> ![Untitled drawing(1)](https://user-images.githubusercontent.com/1891958/122268372-5c1f7c80-ceb2-11eb-9971-a4228953c7d1.png)
> 
> Then select to view the alerts index
> 
> ![image](https://user-images.githubusercontent.com/1891958/122268669-bfa9aa00-ceb2-11eb-8263-71d56cab1b76.png)
> 
> Finally in the right panel you should see the alert.
> 
> ### Configure Wazuh app on the IT server to connect to the Wazuh manager API on OT
> You may have noticed that the alerts are not yet displayed on the `wazuh app` but on `kibana->discover`. This is because we have not yet added to the wazuh app, from the IT server, the connection to the wazuh manager in OT. For this, you must edit the APP configuration file on the IT server (`/usr/share/kibana/optimize/wazuh/config/wazuh.yml`).
> 
> You could add the connection to the Manager API in OT, the file should look like the following:
> 
> ```yaml
> hosts:
>   - managerIT:
>      url: https://localhost
>      port: 55000
>      username: wazuh-wui
>      password: wazuh-wui
>   - managerOT:
>      url: https://192.168.101.10
>      port: 55000
>      username: wazuh-wui
>      password: wazuh-wui
> ```
> 
> Then, when using the wazuh application, you can select which API you want to use, this will allow you to switch between boards of the different managers.
> 
> ![Untitled drawing(1)](https://user-images.githubusercontent.com/1891958/122271434-c980dc80-ceb5-11eb-81bd-5cb90be9ef49.jpg)
> 
> I hope you find this useful, at least informative :).
> 
> Regards, Julian

Hello guys, 
Hope you are good . 
The problem is that when your turn off wazuh_OT , the alerts comming from wazuh_OT are no longer available on the wazuh_it
server

**Comment by juliancnn** [comment]
Hi @MalekBerrich 

This is the expected behavior, since wazuh_OT is the "owner" of its API, and it is the one who has to generate new alerts to store in `alerts.json`.

If you only want to see wazuh_OT's past alerts, which have already been indexed, you can view them in the Discover tab: 
`Kibana -> Discover (in IT host)`:

![image](https://user-images.githubusercontent.com/1891958/203324001-6b233ba2-8a3e-438d-a250-1eba8a821159.png)

**Comment by matthz22** [comment]
> Hi @Serva95,
> 
> Glad to hear you found a solution!
> 
> There is another alternative. You might want to know. You currently have a architecture like the following:
> 
> ![Untitled drawing](https://user-images.githubusercontent.com/1891958/122244521-54081280-ce9b-11eb-9f4c-73fa30c81b92.jpg)
> 
> #### Reviewing the operation of the Wazuh environment
> Wazuh Manager, receives logs from different sources. The logs are processed by Wazuh-Analysisd, and in the case of generating alerts, these are stored in the `alert.json` file. Filebeat, watches the `alert.json` file, reads the alerts and sends them to Elasticsearch. Finally, Wazuh-App (application inside Kibana), is able to communicate with the manager through its API and also to collect alert data from Elasticsearch.
> 
> This is a brief summary, abstracting from details, in order to explain the workaraund (You can learn more about the architecture and operation of wazuh [here](https://documentation.wazuh.com/current/getting-started/architecture.html)).
> 
> ### Workaround
> Another possible solution is to use an architecture like the following:
> 
> ![draw2](https://user-images.githubusercontent.com/1891958/122246160-b44b8400-ce9c-11eb-83ee-25ab5513e9d5.jpg)
> 
> This solution is based on the use of a second instance of Filebeat, which reads the alerts and sends them directly to the Elasticsearch on the other host.
> 
> Advantages:
> 
> * No loss of log source information (location field).
> * Logs are processed only once, with the OT Manager ruleset.
> * Logs are processed in a single Manager,
> 
> Disadvantages:
> 
> * It is a bit more complex than the script that forwards the logs.
> 
> The solution could be carried out in 3 stages:
> 
> 1. Configure Elasticsearch in IT to accept connections from remote hosts.
> 2. Configure a second instance of Filebeat on the OT server
> 3. Configure Wazuh app on the IT server to connect to the Wazuh manager API on OT
> 
> ### Configure Elasticsearch in IT to accept connections from remote hosts.
> If you have the default configuration, elasticsearch will only accept connections from the local host. To change this behavior you should edit the `/etc/elasticsearch/elasticsearch.yml` file and replace the line:
> 
> ```
> network.host: 127.0.0.1
> ```
> 
> with:
> 
> ```
> network.host: 0.0.0.0
> ```
> 
> Then restart the service:
> 
> ```
> systemctl restart elasticsearch.service
> ```
> 
> > you could read more about it [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-network.html#common-network-settings)
> 
> ### Configure a second instance of Filebeat on the OT server
> For this, we will create empty folders for the new instance to use and then clone the current instance filebealbeat
> 
> ```
> # Create empty folders for the new instance
> mkdir  /var/lib/filebeat-remote
> mkdir  /var/log/filebeat-remote
> # Clone the filebeat instance
> cp -a /etc/filebeat /etc/filebeat-remote
> ```
> 
> #### Configuration of the new instance (remote Filebeat)
> To do this, we edit the `/etc/filebeat-remote/filebeat.yml` configuration file, update the certificate path, point it to the IT elastic and disable certificate verification.
> 
> ```
> vim /etc/filebeat-remote/filebeat.yml
> ```
> 
> It could look like the following:
> 
> ```yaml
> # Wazuh - Filebeat configuration file
> output.elasticsearch:
>   # IP of the IT server (where we will send the OT alerts)
>   hosts: ["192.168.0.35:9200"]
>   protocol: https
>   # User and password for elasticsearch, by default in the installation is admin:admin, 
>   # I recommend to change it once everything is working correctly.
>   username: "admin"
>   password: "admin"
>   # Add this option to disable the verification of certificates (This does not disable encryption).
>   ssl.verification_mode: none
>   # Change the certificates path from /etc/filebeat/ to /etc/filebeat-remote/
>   ssl.certificate_authorities:
>     - /etc/filebeat-remote/certs/root-ca.pem
>   ssl.certificate: "/etc/filebeat-remote/certs/filebeat.pem"
>   ssl.key: "/etc/filebeat-remote/certs/filebeat.key"
> setup.template.json.enabled: true
> setup.template.json.path: '/etc/filebeat/wazuh-template.json'
> setup.template.json.name: 'wazuh'
> setup.ilm.overwrite: true
> setup.ilm.enabled: false
> 
> filebeat.modules:
>   - module: wazuh
>     alerts:
>       enabled: true
>     archives:
>       enabled: false
> ```
> 
> Finally, it is necessary to copy the certificates from the IT server to the OT server. For this we copy the following files from the IT server to OT server:
> 
> origin	destination
> IT:/etc/filebeat/certs/root-ca.pem	OT:/etc/filebeat-remote/certs/root-ca.pem
> IT:/etc/filebeat/certs/filebeat.pem	OT:/etc/filebeat-remote/certs/filebeat.pem
> IT:/etc/filebeat/certs/filebeat.key	OT:/etc/filebeat-remote/certs/filebeat.key
> #### Configuration of the service (remote Filebeat)
> Now, we only have to configure the startup of the filebeat-remote as service. To do this, we start from the configuration of the instance already installed, cloning its service script
> 
> ```
> # Clone the service script
> cp /lib/systemd/system/filebeat.service /lib/systemd/system/filebeat-remote.service
> ```
> 
> Then we edit the script, changing some `/etc/filebeat/` paths to `/etc/filebeat-remote/`. It is important to note that the home path does not change, so it should look like the following:
> 
> ```
> [Unit]
> Description=Filebeat sends log files to Logstash or directly to Elasticsearch.
> Documentation=https://www.elastic.co/products/beats/filebeat
> Wants=network-online.target
> After=network-online.target
> 
> [Service]
> 
> Environment="BEAT_LOG_OPTS="
> Environment="BEAT_CONFIG_OPTS=-c /etc/filebeat-remote/filebeat.yml"
> Environment="BEAT_PATH_OPTS=--path.home /usr/share/filebeat --path.config /etc/filebeat-remote --path.data /var/lib/filebeat-remote --path.logs /var/log/filebeat-remote"
> ExecStart=/usr/share/filebeat/bin/filebeat --environment systemd $BEAT_LOG_OPTS $BEAT_CONFIG_OPTS $BEAT_PATH_OPTS
> Restart=always
> 
> [Install]
> WantedBy=multi-user.target
> ```
> 
> Then we reload the list of services and start our second instance of Filbeat:
> 
> ```
> systemctl daemon-reload
> systemctl start filebeat-remote
> systemctl enable filebeat-remote
> ```
> 
> At this point, the **new alerts** generated in OT should arrive at the elasticsearch instance in IT.
> 
> Before proceeding further, we should check if this worked correctly. To do this, forces the generation of an alert on the OT server, e.g. with failed ssh attempt, then go to `Kibana -> Discover` (in IT host)
> 
> ![Untitled drawing(1)](https://user-images.githubusercontent.com/1891958/122268372-5c1f7c80-ceb2-11eb-9971-a4228953c7d1.png)
> 
> Then select to view the alerts index
> 
> ![image](https://user-images.githubusercontent.com/1891958/122268669-bfa9aa00-ceb2-11eb-8263-71d56cab1b76.png)
> 
> Finally in the right panel you should see the alert.
> 
> ### Configure Wazuh app on the IT server to connect to the Wazuh manager API on OT
> You may have noticed that the alerts are not yet displayed on the `wazuh app` but on `kibana->discover`. This is because we have not yet added to the wazuh app, from the IT server, the connection to the wazuh manager in OT. For this, you must edit the APP configuration file on the IT server (`/usr/share/kibana/optimize/wazuh/config/wazuh.yml`).
> 
> You could add the connection to the Manager API in OT, the file should look like the following:
> 
> ```yaml
> hosts:
>   - managerIT:
>      url: https://localhost
>      port: 55000
>      username: wazuh-wui
>      password: wazuh-wui
>   - managerOT:
>      url: https://192.168.101.10
>      port: 55000
>      username: wazuh-wui
>      password: wazuh-wui
> ```
> 
> Then, when using the wazuh application, you can select which API you want to use, this will allow you to switch between boards of the different managers.
> 
> ![Untitled drawing(1)](https://user-images.githubusercontent.com/1891958/122271434-c980dc80-ceb5-11eb-81bd-5cb90be9ef49.jpg)
> 
> I hope you find this useful, at least informative :).
> 
> Regards, Julian

Hey, dude, what's up?
I hope so

I need help, if you can I would appreciate it...

I made all the settings you mentioned above and my wazuh IT only shows the agent, no log is sent... would you be able to help me with this?
